captum/optim/_param/image/images.py

```diff
@@ -1,538 +1,457 @@
-                    if init.dim() == 4
-                    else init.refine_names("C", "H", "W")
-                )
-                init = self.decorrelate(init, inverse=True).rename(None)
-            if squash_func is None:
-                squash_func: Callable[[torch.Tensor], torch.Tensor] = lambda x: x.clamp(
-                    0, 1
-                )
-        else:
-            if squash_func is None:
-                squash_func: Callable[
-                    [torch.Tensor], torch.Tensor
-                ] = lambda x: torch.sigmoid(x)
-        self.squash_func = squash_func
-        self.parameterization = parameterization(
-            size=size, channels=channels, batch=batch, init=init
-        )
-
-    def forward(self) -> torch.Tensor:
-        image = self.parameterization()
-        if self.decorrelate is not None:
-            image = self.decorrelate(image)
-        image = image.rename(None)  # TODO: the world is not yet ready
-        return CudaImageTensor(self.squash_func(image))
+from copy import deepcopy
+from types import MethodType
+from typing import Callable, List, Optional, Tuple, Union
+
+import matplotlib.pyplot as plt
+import numpy as np
+import requests
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+
+try:
+    from PIL import Image
+except (ImportError, AssertionError):
+    print("The Pillow/PIL library is required to use Captum's Optim library")
+
+from captum.optim._param.image.transform import SymmetricPadding, ToRGB
+
+
+class ImageTensor(torch.Tensor):
+    @staticmethod
+    def __new__(cls, x: Union[List, np.ndarray, torch.Tensor] = [], *args, **kwargs):
+        if isinstance(x, torch.Tensor) and x.is_cuda:
+            x.show = MethodType(cls.show, x)
+            x.export = MethodType(cls.export, x)
+            return x
+        else:
+            return super().__new__(cls, x, *args, **kwargs)
+
+    @classmethod
+    def open(cls, path: str, scale: float = 255.0):
+        if path.startswith("https://") or path.startswith("http://"):
+            response = requests.get(path, stream=True)
+            img = Image.open(response.raw)
+        else:
+            img = Image.open(path)
+        img_np = np.array(img.convert("RGB")).astype(np.float32)
+        return cls(img_np.transpose(2, 0, 1) / scale)
+
+    def __repr__(self) -> str:
+        prefix = "ImageTensor("
+        indent = len(prefix)
+        tensor_str = torch._tensor_str._tensor_str(self, indent)
+        suffixes = []
+        if self.device.type != torch._C._get_default_device() or (
+            self.device.type == "cuda"
+            and torch.cuda.current_device() != self.device.index
+        ):
+            suffixes.append("device='" + str(self.device) + "'")
+        return torch._tensor_str._add_suffixes(
+            prefix + tensor_str, suffixes, indent, force_newline=self.is_sparse
+        )
+
+    def show(self, scale: float = 255.0) -> None:
```

Comment:
In light of how often `show` and `export` are needed outside of `ImageTensor`, it might make sense to have both class functions call standalone `show` and `export` functions. 

Are we making this change in this PR or a future one?

Reply:
Let's make this change in this PR. I'll update this again.

Reply:
Do we want to define them outside the classes in the `image.py` file ?
It looks like these two methods have common logic that we can reuse.

Reply:
Yup I'm defining the functions in `captum/optim/_utils/image/common.py` for now. Let me know if there's a better place to put this.